### PR TITLE
[SPARK-46189][PS][SQL] Perform comparisons and arithmetic between same types in various Pandas aggregate functions to avoid interpreted mode errors

### PR DESCRIPTION
**Pull request by bersprockets**
### What changes were proposed in this pull request?

In various Pandas aggregate functions, remove each comparison or arithmetic operation between `DoubleType` and `IntergerType` in `evaluateExpression` and replace with a comparison or arithmetic operation between `DoubleType` and `DoubleType`.

Affected functions are `PandasStddev`, `PandasVariance`, `PandasSkewness`, `PandasKurtosis`, and `PandasCovar`.

### Why are the changes needed?

These functions fail in interpreted mode. For example, `evaluateExpression` in `PandasKurtosis` compares a double to an integer:
```
If(n < 4, Literal.create(null, DoubleType) ...
```
This results in a boxed double and a boxed integer getting passed to `SQLOrderingUtil.compareDoubles` which expects two doubles as arguments. The scala runtime tries to unbox the boxed integer as a double, resulting in an error.

Reproduction example:
```
spark.sql("set spark.sql.codegen.wholeStage=false")
spark.sql("set spark.sql.codegen.factoryMode=NO_CODEGEN")

import numpy as np
import pandas as pd

import pyspark.pandas as ps

pser = pd.Series([1, 2, 3, 7, 9, 8], index=np.random.rand(6), name="a")
psser = ps.from_pandas(pser)

psser.kurt()
```
See Jira (SPARK-46189) for the other reproduction cases.

This works fine in codegen mode because the integer is already unboxed and the Java runtime will implictly cast it to a double.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New unit tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/aggregate/DeclarativeAggregateEvaluator.scala [hunk 1]
```diff
@@ -17,24 +17,24 @@
 package org.apache.spark.sql.catalyst.expressions.aggregate
 
 import org.apache.spark.sql.catalyst.InternalRow
-import org.apache.spark.sql.catalyst.expressions.{Attribute, JoinedRow, SafeProjection}
```
Comment (bersprockets) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/aggregate/DeclarativeAggregateEvaluator.scala:20]: `SafeProjection` works fine until the aggregation buffer contains more than one field. In that case, if the second expression in the function's `updateExpressions` looks back at the buffer's first field, that first field may have already be changed by the first expression.

`MutableExpression`, on the other hand, keeps an intermediate results row, so the current buffer is not affected until all expressions in `updateExpressions` have been evaluated.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala [hunk 2]
```diff
@@ -425,9 +425,9 @@ case class PandasKurtosis(child: Expression)
   override protected def momentOrder = 4
 
   override val evaluateExpression: Expression = {
-    val adj = ((n - 1) / (n - 2)) * ((n - 1) / (n - 3)) * 3
```
Comment (bersprockets) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala:428]: Neither Pandas nor math are my string suits, so I am sort of flying by the seat of my pants here. Nice if someone double checked me.

**Comment by bersprockets** [comment]
@zhengruifeng @HyukjinKwon Probably low priority as someone is unlikely to run into this (unless a code-generated function failed to compile, I suppose), but here's a proposed fix anyway.

**Review by HyukjinKwon — APPROVED** [review]
LGTM but let me defer to @zhengruifeng

**Review by zhengruifeng — APPROVED** [review]
thanks for fixing this

**Comment by zhengruifeng** [comment]
thanks, merged to master/branch-3.5/branch-3.4

**Pull request closed without merging**